builder/window_builder.sh

```diff
@@ -92,6 +92,13 @@ build_window_format() {
   echo "$final_window_format"
 }
 
+prepend_separator() {
+  local field="$1"
+  local separator="${2:- }"
+
+  echo "${field:+$separator$field}"
+}
+
```

Comment:
This function can just be 1 line, but would prefer to just skip the function and inline it everywhere.

Reply:
> This function can just be 1 line, 

Yes, it could be. Something like this:

```bash
echo "${1:+${2:- }$1}"
```

But this way the function seems clearer to me and easier to understand its purpose. I was actually wondering if it could be reused in other places too? But for that it would have to be in some generic script, I was not sure where is the best to put it...

> but would prefer to just skip the function and inline it everywhere.

I was thinking about it too. It would save a couple of lines of code, but the code would be less maintainable as the result, IMO. 

Having an additional function is probably also a tiiiiny bit slower, but not sure anyone would ever notice.

Reply:
Why do you need that `$2` it's never provided it's _always_ going to be a space.

Reply:
In this case - yes. But I was wondering if this function could be reused elsewhere, where the separator could be configurable ...

Reply:
You can always add it in the future without changing current behavior.